### PR TITLE
refactor: remove required from getMember

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
**Please describe the changes this PR makes and why it should be merged:**
Removed `required` from `CommandInteractionOptionResolver#getMember`

In all other types, we can set an option to require. That's why we know that this value will be present or not when we are writing the code. But `getMember` is totally different. In this case, discord neither provide a `MEMBER` type nor a way to limit `USER` type to member only. That's why if we set a `USER` option which is required then also `getMember` can be null as in `USER` type we can provide a user Id of any user who aren't present in that guild.

Here's a example

```ts
    {
      name: "user",
      description: "The user to play against",
      required: true,
      type: "USER",
    }
```
Here the option is required. But if I use. 

```ts
const member = options.getMember("user", true)
```

Then, as this input can be any user who aren't even a member of that guild, so this line of code will through error.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: src/structures/CommandInteractionOptionResolver.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ class CommandInteractionOptionResolver {
    * Gets an option by name and property and checks its type.
    * @param {string} name The name of the option.
    * @param {ApplicationCommandOptionType} type The type of the option.
-   * @param {string[]} properties The properties to check for for `required`.
+   * @param {string[]} properties The properties to check for `required`.
```
Comment (ImRodry) [inline src/structures/CommandInteractionOptionResolver.js:88]: This is correct as-is, doesn’t need to be changed and is also out of the scope of the PR

File: src/structures/CommandInteractionOptionResolver.js [hunk 2]
```diff
@@ -196,12 +196,11 @@ class CommandInteractionOptionResolver {
   /**
    * Gets a member option.
    * @param {string} name The name of the option.
-   * @param {boolean} [required=false] Whether to throw an error if the option is not found.
    * @returns {?(GuildMember|APIGuildMember)}
-   * The value of the option, or null if not set and not required.
+   * The value of the option, or null if the user is not present in the guild or not set and not required.
```
Comment (ImRodry) [inline src/structures/CommandInteractionOptionResolver.js:200]:
```suggestion
   * The value of the option, or null if the user is not present in the guild or the option is not set.
```

File: typings/index.d.ts [hunk 3]
```diff
@@ -49,10 +48,10 @@ import {
   RESTPostAPIApplicationCommandsJSONBody,
   Snowflake,
 } from 'discord-api-types/v9';
+import { Response } from 'node-fetch';
 import { ChildProcess } from 'node:child_process';
 import { EventEmitter } from 'node:events';
 import { AgentOptions } from 'node:https';
-import { Response } from 'node-fetch';
```
Comment (ImRodry) [inline typings/index.d.ts:55]: Sure…?

File: typings/index.d.ts [hunk 4]
```diff
@@ -5463,8 +5460,6 @@ export interface ShardingManagerOptions {
   execArgv?: string[];
 }
 
-export { Snowflake };
-
```
Comment (ImRodry) [inline typings/index.d.ts:5467]: Please don’t add miscellaneous changes to your PR

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: typings/index.d.ts (hunk 3 above)
Comment (imranbarbhuiya) [reply typings/index.d.ts:55]: It was done by ts organize import which organized it alphabetically. You can check it's preset in line 51. Idk which convention do you use. Should I turn that off?

File: typings/index.d.ts (hunk 4 above)
Comment (imranbarbhuiya) [reply typings/index.d.ts:5467]: This line is also moved to line 648

Comment (ImRodry) [reply typings/index.d.ts:5467]: No clue what that is but it doesn’t sound relevant to this PR

Comment (imranbarbhuiya) [reply typings/index.d.ts]: reverted

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Comment by imranbarbhuiya** [comment]
I need to remove some more types from this `test-d.ts` which were removed from main branch. Waiting for #7215 to merge then i can commit the changes.

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**